File: Examples/VisualBasic/Tags/GetTagDetails.vb
Imports System.IO
Imports Aspose.Note
Imports System.Collections.Generic

Namespace Tags
    Public Class GetTagDetails
        Public Shared Sub Run()
            ' ExStart:GetTagDetails
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_Tags()

            ' Load the document into Aspose.Note.
            Dim oneFile As New Document(dataDir & Convert.ToString("TagFile.one"))

            ' Get all RichText nodes
            Dim nodes As IList(Of RichText) = oneFile.GetChildNodes(Of RichText)()

            ' Iterate through each node
            For Each richText As RichText In nodes
                For Each tag In richText.Tags
                    If TypeOf tag Is NoteTag Then
                        Dim noteTag As NoteTag = DirectCast(tag, NoteTag)
                        ' Retrieve properties
                        Console.WriteLine("Completed Time: " + noteTag.CompletedTime.ToString())
                        Console.WriteLine("Create Time: " + noteTag.CreationTime.ToString())
                        Console.WriteLine("Font Color: " + noteTag.FontColor.ToString())
                        Console.WriteLine("Status: " + noteTag.Status.ToString())
                        Console.WriteLine("Label: " + noteTag.Label)
                        Console.WriteLine("Icon: " + noteTag.Icon.ToString())
                        Console.WriteLine("High Light: " + noteTag.Highlight.ToString())
                    End If
                Next
            Next

            ' ExEnd:GetTagDetails
        End Sub
    End Class
End Namespace

File: Examples/CSharp/Tags/GetTagDetails.cs

using System.IO;
using Aspose.Note;
using System;
using System.Collections.Generic;

namespace Aspose.Note.Examples.CSharp.Tags
{
    using System.Linq;

    public class GetTagDetails
    {
        public static void Run()
        {
            // ExStart:GetTagDetails
            // ExFor:ITag
            // ExFor:ITag.CompletedTime
            // ExFor:ITag.CreationTime
            // ExFor:ITag.Icon
            // ExFor:ITag.Label
            // ExFor:ITag.Status
            // ExFor:INoteTag
            // ExFor:INoteTag.FontColor
            // ExFor:INoteTag.Highlight
            // ExFor:NoteTag
            // ExFor:RichText
            // ExFor:RichText.Text
            // ExFor:RichText.ParagraphStyle
            // ExSummary:Shows how to access details of a tag.

            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_Tags();

            // Load the document into Aspose.Note.
            Document oneFile = new Document(dataDir + "TagFile.one");

            // Get all RichText nodes
            IList<RichText> nodes = oneFile.GetChildNodes<RichText>();

            // Iterate through each node
            foreach (RichText richText in nodes)
            {
                var tags = richText.Tags.OfType<NoteTag>();
                if (tags.Any())
                {
                    Console.WriteLine($"Text: {richText.Text}");
                    foreach (var noteTag in tags)
                    {
                        // Retrieve properties
                        Console.WriteLine($"    Completed Time: {noteTag.CompletedTime}");
                        Console.WriteLine($"    Create Time: {noteTag.CreationTime}");
                        Console.WriteLine($"    Font Color: {noteTag.FontColor}");
                        Console.WriteLine($"    Status: {noteTag.Status}");
                        Console.WriteLine($"    Label: {noteTag.Label}");
                        Console.WriteLine($"    Icon: {noteTag.Icon}");
                        Console.WriteLine($"    High Light: {noteTag.Highlight}");
                    }
                }
            }

            // ExEnd:GetTagDetails
        }
    }
}